tests/Api/EncodeTest.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace League\Glide\Api;
+
+use Intervention\Image\Encoders\MediaTypeEncoder;
+use Intervention\Image\ImageManager;
+use Intervention\Image\Interfaces\EncodedImageInterface;
+use Intervention\Image\Interfaces\ImageInterface;
+use Mockery;
+use Mockery\Mock;
+use PHPUnit\Framework\TestCase;
+
+class EncodeTest extends TestCase
+{
+    /**
+     * @var Encode
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $encoder;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $jpg;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $png;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $gif;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $tif;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $webp;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $avif;
+    /**
+     * @var ImageInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $heic;
+
+    public function setUp(): void
+    {
+        $manager = ImageManager::gd();
+        $this->jpg = $manager->read(
+            $manager->create(100, 100)->encode(new MediaTypeEncoder('image/jpeg'))->toFilePointer()
+        );
+        $this->png = $manager->read(
+            $manager->create(100, 100)->encode(new MediaTypeEncoder('image/png'))->toFilePointer()
+        );
+        $this->gif = $manager->read(
+            $manager->create(100, 100)->encode(new MediaTypeEncoder('image/gif'))->toFilePointer()
+        );
+
+        if (function_exists('imagecreatefromwebp')) {
+            $this->webp = $manager->read(
+                $manager->create(100, 100)->encode(new MediaTypeEncoder('image/webp'))->toFilePointer()
+            );
+        }
+
+        if (function_exists('imagecreatefromavif')) {
+            $this->avif = $manager->read(
+                $manager->create(100, 100)->encode(new MediaTypeEncoder('image/avif'))->toFilePointer()
+            );
+        }
+
+        $this->encoder = new Encode();
+    }
+
+    public function tearDown(): void
+    {
+        \Mockery::close();
+    }
+
+    public function testCreateInstance(): void
+    {
+        /**
+         * @psalm-suppress ArgumentTypeCoercion
+         */
+        $this->assertInstanceOf('League\Glide\Api\Encode', $this->encoder);
```

Comment:
```suggestion
        $this->assertInstanceOf(Encode:class, $this->encoder);
```
with `use` statement for it.